<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1440</width>
    <height>810</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(18, 18, 18);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>781</width>
      <height>671</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(43, 43, 43);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="registerButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Registers</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cacheButton">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>10</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Cache</string>
     </property>
    </widget>
    <widget class="QPushButton" name="memoryButton">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>10</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Memory</string>
     </property>
    </widget>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>781</width>
       <height>631</height>
      </rect>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>779</width>
        <height>629</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="2" column="0">
        <widget class="QLabel" name="tableLabelBot">
         <property name="styleSheet">
          <string notr="true">color: rgb(255, 255, 255);</string>
         </property>
         <property name="text">
          <string>Instruction Cache</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="tableLabel1">
         <property name="styleSheet">
          <string notr="true">color: rgb(255, 255, 255);</string>
         </property>
         <property name="text">
          <string>Data Cache</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTableWidget" name="tableWidget">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(111, 111, 111);
alternate-background-color: rgb(133, 133, 133);</string>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QTableWidget" name="tableWidgetBot">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(111, 111, 111);
alternate-background-color: rgb(133, 133, 133);</string>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QPushButton" name="pipelineButton">
     <property name="geometry">
      <rect>
       <x>490</x>
       <y>10</y>
       <width>151</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Pipeline</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>-10</y>
      <width>1601</width>
      <height>101</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(43, 43, 43);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="Title">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>1431</width>
       <height>71</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <pointsize>22</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>RISC-V Based ISA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="TitleOurNames">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>1431</width>
       <height>71</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
background-color: rgba(255, 255, 255, 0);</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="text">
      <string>By Sijia Hao and Karam Kofahi</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>420</y>
      <width>321</width>
      <height>351</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(43, 43, 43);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>321</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Config</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="cache1Or2Way">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Direct Mapped</string>
     </property>
    </widget>
    <widget class="QPushButton" name="step1Cycle">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>101</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Step 1 Cycle</string>
     </property>
    </widget>
    <widget class="QPushButton" name="stepNCycles">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>101</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Step N Cycles</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="numCycles">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>71</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="maximum">
      <number>99999</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pipelineEnabled">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>260</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Pipeline Enabled [x]</string>
     </property>
    </widget>
    <widget class="QPushButton" name="cacheEnabled">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>230</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Cache Enabled [x]</string>
     </property>
    </widget>
    <widget class="QPushButton" name="storeInstructions">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>230</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Save Program Status</string>
     </property>
    </widget>
    <widget class="QPushButton" name="stepToEnd">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Step to end</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>200</y>
       <width>141</width>
       <height>28</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <item>
      <property name="text">
       <string>Load Program</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Program 1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Program 2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Program 3</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="resetState">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>310</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Reset</string>
     </property>
    </widget>
    <widget class="QPushButton" name="confirm">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>141</width>
       <height>29</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Confirm Settings</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_4">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>100</y>
      <width>301</width>
      <height>671</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(43, 43, 43);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>281</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Program</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QScrollArea" name="scrollArea_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>301</width>
       <height>631</height>
      </rect>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="instructionsScroll">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>299</width>
        <height>629</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QTableWidget" name="instructionsTable">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(111, 111, 111);
alternate-background-color: rgb(133, 133, 133);</string>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_5">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>100</y>
      <width>321</width>
      <height>311</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(43, 43, 43);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="cycleLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Cycle: 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="PCLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Program Counter: 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="memAccessLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Memory Accesses: 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="cacheHitsLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Cache Hits: 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="cacheMissesLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(43, 43, 43);
color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string>Cache Misses: 0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
